Remove deprecated wasm type check

backes opened the pull request:
This removes uses of the "IsWebAssemblyCompiledModule" method, which is
deprecated in V8 v8.1 and will be removed in v8.2.
We could replace it by "IsWasmModuleObject", but since it's unused in
node (C++) anyway, and there are better ways to check for wasm modules in js,
I just remove the definition.

backes commented:
Not sure what the exact deprecation policy is. Can you check if this is the correct way please?
Also, is version 14 the right version to put in the deprecation note?

addaleax commented:
> Not sure what the exact deprecation policy is. Can you check if this is the correct way please?

A documentation-only deprecation is fine, but it only applies to cases in which the method itself is still there. If `IsWasmModuleObject` also works, it would probably be best to either make them aliases of each other, or if that doesn’t work, make `isWasmModuleObject()` be a stub that just returns e.g. `false` always (and then document that fact).

I’d probably also leave the documentation for the method in the same place – I know it’s a bit odd that there’s a separate section for deprecated APIs where this would seem to fit in well, but I think it makes sense to keep it under the `util.types` heading in this case (all the other APIs listed as deprecated are top-level functions on the `util` module).

> Also, is version 14 the right version to put in the deprecation note?

We use `REPLACEME` as a placeholder that gets replaced automatically as part of the release process. :)

addaleax commented:
That being said, in this repository anything that works for you is fine, I guess. My reply only applies for how we’d handle this situation in the main nodejs/node repository.

backes commented:
Thanks, Anna! I moved the documentation back, and replaced the version by `REPLACEME`.
We could still implement this method, or add another "isWasmModuleObject" method instead, but is that worth it, given that you can just do a standard `instanceof` check?

backes commented:
@addaleax Can you take a look please at the last version? If it looks fine, feel free to merge, as I don't have permissions for that.
Thanks!
